src/main/java/org/fogbowcloud/arrebol/ArrebolController.java

```diff
@@ -118,13 +144,21 @@ public void start() {
         //commit the job pool to DB using a COMMIT_PERIOD_MILLIS PERIOD between successive commits
         //(I also specified the delay to the start the fist commit to be COMMIT_PERIOD_MILLIS)
         this.jobDatabaseCommitter.schedule(new TimerTask() {
-                    public void run() {
-                        LOGGER.info("Commit job pool to the database");
-                        for(Job job : jobPool.values()) {
-                            jobRepository.save(job);
-                        }
-                    }
-                }, COMMIT_PERIOD_MILLIS, COMMIT_PERIOD_MILLIS
+                                               public void run() {
+                                                   LOGGER.info("Commit job pool to the database");
+                                                   jobRepository.save(jobPool.values());
+                                               }
+                                           }, COMMIT_PERIOD_MILLIS, COMMIT_PERIOD_MILLIS
+        );
+
+        this.jobStateMonitor.schedule(new TimerTask() {
+                                          public void run() {
+                                              LOGGER.info("Updating job states");
+                                              for (Job job : jobPool.values()) {
+                                                  updateJobState(job);
+                                              }
+                                          }
+                                      }, UPDATE_PERIOD_MILLIS, UPDATE_PERIOD_MILLIS
```

Comment:
does anyone know what is this "read from bd" TODO?

Reply:
I believe that when the job states are updated, instead of reading the map jobs, read the database jobs

Reply:
do you think this modification is necessary? If so, update the TODO mark